win32/sendmail.c

```diff
@@ -978,6 +1009,46 @@ static unsigned long GetAddr(LPSTR szHost)
 	return (lAddr);
 } /* end GetAddr() */
 
+/* returns the contents of an angle-addr (caller needs to efree) or NULL */
+static char *get_angle_addr(char *address)
+{
+	zend_bool in_quotes = 0;
+	char *p1 = address, *p2;
+
+	while ((p1 = strpbrk(p1, "<\"\\")) != NULL) {
+		if (*p1 == '\\' && in_quotes) {
+			if (p1[1] == '\0') {
+				/* invalid address; let SMTP server deal with it */
+				return NULL;
+			}
+			p1++;
+		} else if (*p1 == '\"') {
+			in_quotes = !in_quotes;
+		} else if (!in_quotes) {
```

Comment:
Won't this interpret `\` outside of quotes as `<`?

Reply:
Oh, yes, forgot to update after backslash handling. Thank again!